pkg/catalog/endpoint.go

```diff
@@ -36,3 +36,17 @@ func (mc *MeshCatalog) GetResolvableServiceEndpoints(svc service.MeshService) ([
 	}
 	return endpoints, nil
 }
+
+// ListEndpointsForPodFromServiceAccount retrieves the list of IP addresses for the given service account
+func (mc *MeshCatalog) ListEndpointsForPodFromServiceAccount(sa service.K8sServiceAccount) []endpoint.Endpoint {
```

Comment:
The signature of this function seems a bit confusing because endpoints are for a service and not service accounts, yet this function only takes a service account as input and returns all endpoints for pods in it.

I think it would be much cleaner to change this as follows and keep the logic in EDS simple.
In `catalog`:
`ListAllowedEndpointsForService(downstreamIdentity service.K8sServiceAccount, upstreamSvc service.MeshService) []endpoint.Endpoint`

In `EDS`, use `ListAllowedEndpointsForService(...)` directly without any need for filtering endpoints.